File: general/src/main/java/com/github/ummo93/ModuleConfig.java
package com.github.ummo93;

import com.github.ummo93.framework.Scene;
import com.github.ummo93.game.MainScene;
import com.github.ummo93.game.ai.AiBehaviourStrategy;
import com.github.ummo93.game.ai.ChaserBehaviour;
import com.google.inject.AbstractModule;

public class ModuleConfig extends AbstractModule {
    @Override
    protected void configure() {
        bind(AiBehaviourStrategy.class).to(ChaserBehaviour.class);
        bind(Scene.class).to(MainScene.class).asEagerSingleton();
    }
}

File: sample-game/src/main/kotlin/com/github/ummo93/config/ModuleConfig.kt
package com.github.ummo93.config

import com.github.ummo93.framework.Scene
import com.github.ummo93.game.MainScene
import com.github.ummo93.game.ai.ChaserBehaviour
import com.google.inject.AbstractModule
import com.github.ummo93.game.ai.AiBehaviourStrategy

class ModuleConfig : AbstractModule() {
    override fun configure() {
        bind(AiBehaviourStrategy::class.java).to(ChaserBehaviour::class.java)
        bind(Scene::class.java).to(MainScene::class.java).asEagerSingleton()
    }
}